### DISCUSSION TITLE
Bypass branchConcurrentLimit for one specific dependency?

### DISCUSSION THREAD

**Discussion by screendriver** [renovatebot/renovate#10766] [Request Help]
I have two repositories `A` and `B`. `A` has a lot of npm dependencies and one of them is the package `B`. In repository `A` I have defined `prConcurrentLimit` and `branchConcurrentLimit`. Additionally I set in `packageRules` that `B` should be auto merged (`"automerge": true` and `"automergeType": "branch"`).

Now my problem is that when repository `A` has reached its limits (which happens all the time because it has a lot of dependencies) it does not automatically update the dependency `B` because of the concurrent limits.

Is there a way to introduce an exception (somewhere in `packageRules`?) for `B` so that `B` gets **always** automatically merged without counting into the given limits? The reason for that is that `B` is one of the most important dependencies which should always be updated as soon as possible.

**Comment by viceice** [comment]
nope, you need a higher `branchConcurrentLimit` with a lower `prConcurrentLimit`, so branch automerge can pass.

if you use stabillitydays with `internalChecksFilter=strict`, you can suppress branches until days match

https://docs.renovatebot.com/configuration-options/#internalchecksfilter

**Reply by screendriver** [reply]
> nope, you need a higher branchConcurrentLimit

the main problem here is that when I would do this every branch kicks of a new CI pipeline (also when new commits are coming in) which in our case consumes a lot of CPU and bandwidth resources and blocks other teams in the company. For example when 20 pipelines are running in parallel in repository `A` then repository `C` has to wait until enough runners are free again. That's the reason why we introduced limits for all repositories to reduce the overall noise.

> if you use stabillitydays

`stabilityDays` does not help here because we go the continuous delivery way where every commit to `main` creates a new npm package which should immediately deployed to production.

**Reply by viceice** [reply]
I know you problem, i limit this by using something like

```js
  prCommitsPerRunLimit: 20,
  rebaseWhen: 'auto',
```

So on one renovate run i only allow 20 pushes over all repos. and i only allow renovate to update manual merge pr's to update when they are conflicted.
